booleans.js

```diff
@@ -0,0 +1,124 @@
+function negate(a) {
+  return !a;
+};
+
+function both(a, b) {
+  if (a == true && b == true) {
+    return true;
+  } else {
+    return false;
+  };
+
+};
+
+function either(a, b) {
+  if (a == true | b == true) {
+
+    return true;
+  } else {
+
+    return false;
+  };
+};
+
+function none(a, b) {
+  return negate(either(a, b));
+};
+
+function one(a, b) {
+  if ((a == true | b == true) && a !== b) {
+    return true;
+  } else {
+    return false;
+  };
+};
+
+function truthiness(a) {
+  if (a) {
+    return true;
+  } else {
+    return false;
+  };
+};
+
+function isEqual(a, b) {
+  return a === b;
+};
+
+function isGreaterThan(a, b) {
+  return a > b;
+};
+
+function isLessThanOrEqualTo(a, b) {
+  return a <= b;
+};
+
+function isOdd(a) {
+  if (a % 2 == 1) {
+    return true;
+  } else {
+    return false;
+  };
+};
+
+function isEven(a) {
+  if (a % 2 == 0) {
+    return true;
+  } else {
+    return false;
+  };
+};
+
+function isSquare(a) {
+  if (Number.isInteger(Math.sqrt(a)) == true) {
+    return true;
+  } else {
+    return false;
+  };
+};
+
+function startsWith(char, string) {
+  if (string.charAt(0) == char) {
+    return true;
+  } else {
+    return false;
+  };
+};
+
+function containsVowels(string) {
+  string = string.toLowerCase();
+  const vowels = ['a', 'e', 'i', 'o', 'u'];
+  for (let i = 0; i < string.length; i++) {
+    if (vowels.includes(string[i])) return true;
+  }
+  return false;
+};
```

Comment:
So this feels like a good chance to expand on some use of regex you tried earlier!
This is one of those things that you'll end up using at some point, its a very versatile and powerful way of matching patterns in strings very quickly. In this case we want to use it to define our set of vowels: `/[aeiou]/`.

What this example means is: match any part of a given string with any character between the square brackets. In this case it will only match these exact characters in lower case, but a small modifier of the `i` flag at the end (making it `/[aeiou]/i`) makes it case insensitive, so will match both cases of anything in the brackets.
These are used with the `.match` js function, like this:
```
someString.match(/[aeiou]/i)
```
this will return either true if it finds a match, or null if not. So for this function we could use 
```
  return string.match(/[aeiou]/i);
```
and it would return `true` if the string contained a lower or upper case vowel, and `null` if not. But since we only want a `true` or `false` value to be returned, we can use that double negation (!!) method again to convert the `null` value to false:
```
  return !!string.match(/[aeiou]/i);
```
Way more concise.